pkg/cmd/cli/describe/describer.go

```diff
@@ -118,6 +119,23 @@ func (d *BuildDescriber) DescribeParameters(p buildapi.BuildParameters, out *tab
 	}
 }
 
+func (d *BuildDescriber) DescribeDockerImage(s *buildapi.STIBuildStrategy, namespace string, out *tabwriter.Writer) {
+	if s.Image == "" {
+		c := d.ImageRepositories(namespace)
+		imageRepo, err := c.Get(s.From.Name)
+		if err == nil {
+			formatString(out, "Image Repository", imageRepo.Status.DockerImageRepository)
```

Comment:
Look at DescribeTriggers and specifically the logic for describing an ImageChange trigger for how to handle the From field.

also you need to show the Image field if it's present, and you need to show the Tag field if present.

and for that matter the Scripts field if present.